lib/Dialect/SV/SVOps.cpp

```diff
@@ -172,6 +173,37 @@ MacroRefExprSEOp::getReferencedMacro(const hw::HWSymbolCache *cache) {
   return ::getReferencedMacro(cache, *this, getMacroNameAttr());
 }
 
+//===----------------------------------------------------------------------===//
+// MacroErrorOp
+//===----------------------------------------------------------------------===//
+
+SmallString<32> MacroErrorOp::getMacroIdentifier() {
+  SmallString<32> id;
+  id.append("_ERROR");
+
+  auto msg = getMessage();
+  if (!msg)
+    return id;
+
+  bool split = true;
+  for (auto c : *msg) {
+    if (isalnum(c)) {
+      if (split)
+        id.push_back('_');
+      id.push_back(c);
+      split = false;
+    } else {
+      split = true;
+    }
+  }
```

Comment:
This would be simpler with replacing `llvm::isSpace` with `_` and dropping anything that isn't `llvm::isAlnum`.

I grok the "collapse strings of non-alnum to `_` except the leading non-alnum". However, this may be slightly too clever.

Another direction is to make this verified as being `isSpace` or `isAlnum` in a verifier for the op for additional predictability.

Reply:
LGTM.